### PR TITLE
Replace context aliases

### LINKED ISSUE #152: Replace context aliases
`ReplyCtx` and `IntstantiateCtx` were type aliases of order `MigrateCtx` and `ExecCtx`.
Unfortunately, in case of compilation errors, it points to the original type, which could be misleading.

### PR DESCRIPTION
**Pull request by jawoznia**
Closes #152

### CODE REVIEW

**Review by hashedone — APPROVED** [review]
LGTM

File: MIGRATING.md [hunk 1]
```diff
@@ -2,6 +2,24 @@
...
+   |
+56 |         let result = self.whitelist.instantiate(ctx.branch(), admins, mutable)?;
+   |                                     ----------- ^^^^^^^^^^^^ expected `ExecCtx<'_>`, found `InstantiateCtx<'_>`
+   |                                     |
+   |                                     arguments to this method are incorrect
+   |
+   = note: expected struct `ExecCtx<'_>`
+              found struct `InstantiateCtx<'_>`
+```
+
```
Comment (hashedone) [inline MIGRATING.md:22]: Too long of an entry for changelog. The only description is enough.

File: MIGRATING.md [hunk 2]
```diff
@@ -2,6 +2,24 @@
 
 This guide explains what is needed to upgrade contracts when migrating over major releases of `sylvia`. Note that you can also view the [complete CHANGELOG](https://github.com/CosmWasm/sylvia/blob/main/CHANGELOG.md) to understand the differences.
 
+## 0.5.0 -> Unreleased
+
+- InstantiateCtx and ExecCtx could be previosly used interchangeably. They are currently separate
+  types. The same applies to ReplyCtx/MigrateCtx pair.
```
Comment (hashedone) [inline MIGRATING.md:8]: Use backtics (\`) for code entity names (eg \`InstantiateCtx\`).

**Review by hashedone — COMMENTED** [review]
Changelog fix + please reabase for better commit history.

File: MIGRATING.md (hunk 1 above)
Comment (jawoznia) [reply MIGRATING.md:21]: This is MIGRATING, but applied some changes as discussed offline

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+- ExecCtx and ReplyCtx are no longer type aliases.
+
```
Comment (hashedone) [inline CHANGELOG.md:10]: Let's try to leave it to `release-please` - we will see how it works first. At worse, we will recreate the changelog on release.

**Review by hashedone — APPROVED** [review]
LGTM - one comment to apply, please fix commit history (leave merge commit for now, for future - prefer to rebase over merge).

**Pull request merged**